### PR TITLE
fix: record which matches the current request error. 

### LINKED ISSUE #67: bug: record which matches the current request error
With the following two conditions met:

- opts.method matched routr.methods
- opts.matched = {}

the route match will fail.

case references: apache/apisix#1574

bug causes
https://github.com/api7/lua-resty-radixtree/blob/6b5d65eb9853c91b1d23cc4cea8941be74bb5f20/lib/resty/radixtree.lua#L663-L666
between `match_route_opts` and `compare_param` , should check whether it is resuful style url pass parameter, or prefix matching, the prefix matching does not need to be entered `compare_param`.

### PR DESCRIPTION
**Pull request by tzssangglass**
fix #67

### CODE REVIEW

**Comment by tzssangglass** [comment]
haha, so my CLA signature status doesn't update because I didn't set up eamil locally in git! ^_^|||

**Comment by membphis** [comment]
> haha, so my CLA signature status doesn't update because I didn't set up eamil locally in git! ^_^|||

it is right now. congratulation ^_^

**Review by membphis — COMMENTED** [review]
you can make a try this version, it is simple.

```diff
diff --git a/lib/resty/radixtree.lua b/lib/resty/radixtree.lua
index 9bc7303..fb44b3d 100644
--- a/lib/resty/radixtree.lua
+++ b/lib/resty/radixtree.lua
@@ -451,6 +451,10 @@ local function compare_param(req_path, route, opts)

     local pat, names = fetch_pat(route.path_org)
     log_info("pcre pat: ", pat)
+    if #names == 0 then
+        return true
+    end
+
     local m = re_match(req_path, pat, "jo")
     if not m then
         return false

File: t/matched.t [hunk 1]
```diff
@@ -0,0 +1,145 @@
+# vim:set ft= ts=4 sw=4 et fdm=marker:
+
+use t::RX 'no_plan';
+
+repeat_each(1);
+run_tests();
+
+__DATA__
+=== TEST 1: prefix matching with method, and matched is not nil.
```
[the remark this answers is not in the document]
Comment (tzssangglass) [reply t/matched.t:9]: ok, i try

[the remark this answers is not in the document]
Comment (tzssangglass) [reply t/matched.t:9]: I also felt that the code I wrote was a bit cumbersome.

**Comment by tzssangglass** [comment]
But I'm thinking that if change it as above, the request will go into `compare_param` ->  `fetch_pat`, execute more steps, and the `compare_param` function isn't pure, so is there any way to get just restful style requests into the `compare_param` function?

**Comment by membphis** [comment]
> But I'm thinking that if change it as above, the request will go into `compare_param` -> `fetch_pat`

that is correct. when we enable `opts.matched`, `resty-radixtree` should do this step, it is necessary.

**Review by membphis — APPROVED** [review]

**Comment by membphis** [comment]
@tzssangglass merged, many thx

**Comment by tzssangglass** [comment]
> @tzssangglass merged, many thx

ok, after release the new version, I will use this to fix https://github.com/apache/apisix/issues/1574

**Comment by membphis** [comment]
@tzssangglass I have released a new version `v2.4` , you can continue your job now.  ^_^

**Comment by tzssangglass** [comment]
great！\(‵▽′)/

**Pull request merged**